crates/common/src/ckb_decimal.rs

```diff
@@ -0,0 +1,14 @@
+use gw_types::U256;
+
+// NOTE: u64::MAX is about 10^18, U256::MAX is about 10^77. Multiple 10 pow
+// `CKB_DECIMAL_POW_EXP` should not overflow.
+pub const CKB_DECIMAL_POW_EXP: u32 = 10;
+pub const CKB_DECIMAL_POWER_TEN: u64 = 10u64.pow(CKB_DECIMAL_POW_EXP);
+
+pub fn to_18(amount: u64) -> U256 {
+    U256::from(amount) * CKB_DECIMAL_POWER_TEN
+}
+
+pub fn from_18(amount: U256) -> U256 {
```

Comment:
```suggestion
pub fn from_18(amount: U256) -> u64 {
```

Avoid maintaining two kinds of `U256` with different precisions.

Reply:
Good point, I think we should using a wrapped type to avoid conversion by hand.

```
CKBCapacity(U256);

impl CKBCapacity {
  fn to_layer1() -> u64 {}
  fn to_layer2() -> U256 {}
  fn from_layer1(n: u64) -> Self {}
  fn from_layer2(n: U256) -> Self {}
}
```

This provides more clue on the source code than the current version:

```
CKBCapacity::from_layer1(deposit.capacity()).to_layer2()

// VS

to_18(deposit.capacity())
```